### DISCUSSION TITLE
Trouble when generating CreateSeuratObject from raw count matrix

### DISCUSSION THREAD

**Discussion by treps-l** [satijalab/seurat#6846] [Q&A] [answered]
Hello everyone,

I am facing trouble when generating a Seurat Object from a publicly available raw count matrix of scRNA-seq data.
The matrix can be downloaded from there (https://www.dropbox.com/s/kdf9hvaze3grlnh/Raw%20counts%20data.zip?dl=1) but it is pretty large (100 K cells) so I created a raw count matrix having the same format with only 5 cells and 4 genes (attached).
[Data.csv](https://github.com/satijalab/seurat/files/10359514/Data.csv)

When I run the following code I end up with a Seurat Object but I lost my cell ids on the way and have an extra row. Do you have any idea on how to circumnavigate this? I did check my data frame before generating the Seurat Object, and it looks fine to me. Is this a format issue?

dataMatrix <- read.csv(file = "Data.csv", header = TRUE, sep = ",")
observations <- dataMatrix[[1]]
features <- colnames(dataMatrix)[-1]
dataMatrix2 <- transpose(dataMatrix[, -1])
colnames(dataMatrix2) <- observations
dataMatrix2$Feature <- features
setcolorder(dataMatrix2, c("Feature", observations))
names(dataMatrix2)[1] <- "orig.ident"
dataMatrix2
<img width="371" alt="image" src="https://user-images.githubusercontent.com/122088579/211490374-3f897261-602e-484d-86fe-6f455b290cdd.png">
seuratObject <- CreateSeuratObject(counts = dataMatrix2)
View(seuratObject@meta.data)
<img width="385" alt="image" src="https://user-images.githubusercontent.com/122088579/211225659-9742a4b6-f3e6-499c-85c8-cc53ab0ef41a.png">

Thanks in advance,

**Comment by treps-l** [comment]
Anybody could help on this? I am pretty sure it is formatting issue that could be easily fixed by proficient R users!
Thanks in advance for your help.

**Comment by samuel-marsh** [comment, marked as answer]
Can you post link to public data source repo and not Dropbox?

**Reply by treps-l** [reply]
The original Fastq files can be found here : https://www.ebi.ac.uk/biostudies/arrayexpress/studies/E-MTAB-6308#
However the raw count matrix was generated by the host lab and can only be downloaded using the Dropbox link above. This link is from their platform : https://endotheliomics.shinyapps.io/lung_ectax/
when selecting the "Human Lung" dataset.

**Reply by samuel-marsh** [reply]
Hi,

So at least when I try and download the human data I get message when I try to read it in saying that the .zip file for Data.csv is corrupt.  But the mouse file works so here is how I loaded that.  I had no issues running the following:

```
library(tidyverse) #only need tibble and magrittr if you want to load minimal packages
library(Seurat)

# Load raw and meta data
data <- data.table::fread(unzip("~/Downloads/mouse_raw.zip", "Data.csv"), data.table = F)
meta <- data.table::fread(unzip("~/Downloads/mouse_raw.zip", "Metadata.csv"), data.table = F)

# Data is already in gene (row) by cell (column format) so just move feature names to rownames
data_mod <- data %>% 
  column_to_rownames("Feature")

# Create object
seu <- CreateSeuratObject(counts = data_mod)
View(head(seu@meta.data, 5))

                            orig.ident nCount_RNA nFeature_RNA
ortho_NECAAACCTGCAGATCTGT.1      ortho       1698         1014
ortho_NECAAACCTGGTGATAAAC.1      ortho       3403         1704
ortho_NECAAACCTGTCTGTCTAT.1      ortho       1926         1090
ortho_NECAAACGGGAGAGTGAGA.1      ortho        668          512
ortho_NECAAACGGGCAATCTACG.1      ortho       2874         1443

# Add extra meta data
# First move cell names to rownames
meta_mod <- meta %>% 
  column_to_rownames("Observation")

# Add to object
seu <- AddMetaData(object = seu, metadata = meta_mod)

                            orig.ident nCount_RNA nFeature_RNA   Experiment
ortho_NECAAACCTGCAGATCTGT.1      ortho       1698         1014 Experiment 1
ortho_NECAAACCTGGTGATAAAC.1      ortho       3403         1704 Experiment 1
ortho_NECAAACCTGTCTGTCTAT.1      ortho       1926         1090 Experiment 1
ortho_NECAAACGGGAGAGTGAGA.1      ortho        668          512 Experiment 1
ortho_NECAAACGGGCAATCTACG.1      ortho       2874         1443 Experiment 1
                            Original.name Condition Endothelial.cell
ortho_NECAAACCTGCAGATCTGT.1        API217       NEC              Yes
ortho_NECAAACCTGGTGATAAAC.1        API217       NEC              Yes
ortho_NECAAACCTGTCTGTCTAT.1        API217       NEC              Yes
ortho_NECAAACGGGAGAGTGAGA.1        API217       NEC              Yes
ortho_NECAAACGGGCAATCTACG.1        API217       NEC              Yes
                                        Cluster
ortho_NECAAACCTGCAGATCTGT.1 capillaries type II
ortho_NECAAACCTGGTGATAAAC.1  capillaries type I
ortho_NECAAACCTGTCTGTCTAT.1 capillaries type II
ortho_NECAAACGGGAGAGTGAGA.1          lymphatics
ortho_NECAAACGGGCAATCTACG.1                vein

```

Best,
Sam

**Reply by treps-l** [reply]
Hi Sam,
Many thanks for your answer, it works perfectly now!
Best,
Lucas
